### ISSUE TITLE
Feature request: Find & Replace

### ISSUE THREAD

**Issue by ianaf100** [careteditor/caret#295]
It would be very helpful to have a "find and replace" feature, as it's something I use constantly and have become pretty reliant on. Currently I must open another editor if I need to replace a number of text instances, so it'd be a helpful improvement (for me at least).  Just an idea.  Love the editor so far

**Comment by erusev** [comment]
Thanks, we're working on it, meanwhile, you can use cmd+ctrl+g to select all occurrences and type over to replace

**Comment by ianaf100** [comment]
Didn't realize. Mine was actually set to Alt-F3 by default, but this works great.

**Comment by pslobo** [comment]
I've mentioned this on Twitter already, but if we were able to go from a search straight to cmd+ctrl+g, this would suffice for most Find & Replace functions.

So flow would be:

- cmd+F, type search string
- Press escape
- press cmd+ctrl+g
- type replacement text.

As it stands, you have to type the search term, then move to the mouse to select the first instance and then only cmd+ctrl+g works.

**Comment by ddmng** [comment]
This combination does not work under Linux :(

**Comment by BsNoSi** [comment]
Find & Replace is imo something different in behaviour to described "select solution". This way selects *all* matches an will change *all* maches. Find & Replace (typically) offers an option, to skip a match, because not each match will/shall change.

**Comment by borekb** [comment]
Just wanted to say that I agree with @BsNoSi and @ianaf100, a full-fledged search & replace would be a much-welcome addition. The current workaround is not really usable for me so I'm forced to switch to another editor to do the replace.

**Comment by erusev** [comment]
> The current workaround is not really usable for me so I'm forced to switch to another editor to do the replace.

Could you, please, elaborate on this? I understand the point of @BsNoSi about the case where you want to skip some items, but other than that, what are the benefits of Find and Replace over select all (or multiple) occurrences and type over?

Thanks!

**Comment by borekb** [comment]
Yes, doing it _carefully_ is the main point. In a larger document where I cannot see all occurrences, blindly overwriting everything is not really an option.

**Comment by erusev** [comment]
Sure, but there's also the option of Selection › Add Next Occurrence to manually add occurrences. Wouldn't this serve the same purpose? Thanks!

**Comment by borekb** [comment]
Sorry, I didn't express myself clearly enough. In a typical document more than one page long, it's important for me to:

1. Have confidence that I'm replacing only what should be replaced.
2. Can skip certain occurrences.
3. Can undo specific replaces easily.

Maybe you could add a command like "Skip to Next Occurrence" to support skipping but overall, selection-based search and replace feels quite fragile, e.g., imagine that I make a careful selection but then click somewhere accidentally..

Don't get me wrong, I love multi-cursor editing and it is extremely elegant in certain cases, but additionally to that, I'd like to have the traditional find & replace in Caret.

**Comment by erusev** [comment]
Thanks for the thoughtful response! I'll see what we can do.

**Comment by jmattlandrum** [comment]
Having a standard find/replace is pretty important.
Maybe I am missing something. I am surprised more people aren't complaining. I figured there was some sort of editor API you could use for this.

**Comment by astoilkov** [comment]
@jmattlandrum If you click on the Replace menu item in the application you will see "To replace, use Selection › Select Each to select all occurrences then type.". Have you tried this way of replacing?

**Comment by jmattlandrum** [comment]
I have. I think this has limited use, just slightly better than
search/manual.

I am sure you all have used many editors. You are missing functionality
that nearly all of them have.

**Comment by astoilkov** [comment]
@jmattlandrum We are open to implementing find & replace but currently we consider using multiple cursors a more elegant solution. Let me give you an example:
- using multiple cursors: select word, hit Cmd/Ctrl+Shift+L, type
- using find & replace: select word, hit Cmd+Alt+F(for opening theoretical find & replace), hit Tab to go to replace field, type, hit enter

Do you see how multiple cursors require 3 operations and find & replace require 5?

It would be great to describe a scenario where multiple cursors approach is limited. This would help us understand the problem better. Thanks in advance.

**Comment by rkalis** [comment]
People also use Find/Replace if they want to replace *some but not all* occurrances of a word. E.g. in VS Code you can enter something in the Find box, then the replacement in the Replace box and tab through all occurrances, only replacing the ones you want to replace. For this scenario multiple cursors seems limited.

**Comment by astoilkov** [comment]
@rkalis Yes, indeed, with the current implementation of multiple cursors this limitation exists. In order to fix it, we are considering an improvement that lets you skip the current multiple cursor selection and selecting the next. This feature is supported by Sublime Text for example. We are also considering adding find/replace because this can be complicated for some users.

**Comment by jmattlandrum** [comment]
Just look at MicroSoft Word or VS Code, for example. Virtually everyone is familiar with that Find/Replace model.
